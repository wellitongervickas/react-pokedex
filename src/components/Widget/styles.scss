@import '../../assets/scss/variables';

.widget {
  background-color: $white;
  border-radius: $size-default;
  box-shadow: 0 0 $size-default rgba(0, 0, 0, 0.2);
  color: $black;
  padding: $size-default;
}
