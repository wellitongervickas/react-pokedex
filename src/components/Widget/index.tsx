import React, { ReactNode } from 'react';

import 'components/Widget/styles.scss';

interface WidgetInterface {
  children: ReactNode
}

const Widget = ({ children }: WidgetInterface) => (
  <div className="widget">{children}</div>
);

export default Widget;
