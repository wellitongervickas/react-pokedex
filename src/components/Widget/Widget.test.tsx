import React from 'react';
import { render } from '@testing-library/react';

import Widget from '.';

describe('Component Widget', () => {
  it('renders Widget component', () => {
    const { container, getByText } = render(
      <Widget>
        <span>Hello World</span>
      </Widget>
    );

    const tree = container.firstElementChild;

    expect(getByText('Hello World')).toBeInTheDocument();

    expect(tree?.classList.contains('widget')).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
