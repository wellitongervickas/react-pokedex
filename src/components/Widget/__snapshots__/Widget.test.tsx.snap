// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Widget renders Widget component 1`] = `
<div
  class="widget"
>
  <span>
    Hello World
  </span>
</div>
`;
